File: Excel/691 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/691 Ranking.xlsx"
input = read_excel(path, range = "A2:B15")
test  = read_excel(path, range = "C2:D15")

result = input %>%
  mutate(Rank1 = dense_rank(Score)) %>%
  mutate(Rank2 = row_number()/100 + Rank1, .by = Rank1) %>%
  select(Rank1, Rank2)

all.equal(result, test)
#> [1] TRUE

File: Excel/600-699/691/691 Challenge.py
import pandas as pd
from scipy.stats import rankdata

path = "691 Ranking.xlsx"

input = pd.read_excel(path, usecols="A:B", skiprows=1, nrows=14)
test = pd.read_excel(path, usecols="C:D", skiprows=1, nrows=14)

input['Rank1'] = rankdata(input['Score'], method='dense').astype("int64")
input['Rank2'] = input.groupby('Rank1').cumcount() + 1
input['Rank2'] = input['Rank1'] + input['Rank2'] / 100

result = input[['Rank1', 'Rank2']]

print(result.equals(test)) # True